Sources/Private/Experimental/Layers/SolidLayer.swift

```diff
@@ -33,6 +33,17 @@ final class SolidLayer: CALayer {
     fatalError("init(coder:) has not been implemented")
   }
 
+  /// Called by CoreAnimation to create a shadow copy of this layer
+  /// More details: https://developer.apple.com/documentation/quartzcore/calayer/1410842-init
+  override init(layer: Any) {
+    guard let layer = layer as? Self else {
+      fatalError("init(layer:) incorrectly called with \(type(of: layer))")
+    }
+
+    solidLayer = layer.solidLayer
+    super.init(layer: layer)
+  }
```

Comment:
Seems like we're repeating this logic a number of times, wondering if we could generalize?

Reply:
Yeah I'm really not a fan this layer shadowing API -- the requirements that you have to `override init(layer: Any)` and must call `super.init(layer: layer)` (or else things silently break) makes this tricky to generalize. 

I tried some approaches to generalize this in a way that individual layer subclasses only have to implement a method like:

```swift
func copy(from self: Self) { ... }
```

where the casting is handled automatically by the base implementation

but:
 1. this can't me an instance method, because it has to run before calling `super.init`
 2. this can't be a static copying method of the shape `(Self) -> Self`, because then you can't call `super.init(layer: Any)`.
 3. you also can't implement an `override init` from a protocol extension, so there's no feasible way to provide a default implementation of the init afaict
 
We could potentially extract out the `layer as? Self` casting, but its already pretty lighweight and I couldn't come up with a spelling that felt good enough to be better than a standard `guard`.

Any thoughts or ideas? I would certainly love to improve this if we can think of something.

Reply:
If you investigated and there weren't any clean options to factor out seems totally fine to me to keep as is 👍